File: jdroid-android-sample/src/main/java/com/jdroid/android/sample/ui/uri/UriMapperActivity.java
package com.jdroid.android.sample.ui.uri;

import androidx.fragment.app.Fragment;

import com.jdroid.android.activity.FragmentContainerActivity;
import com.jdroid.android.uri.UriHandler;

public class UriMapperActivity extends FragmentContainerActivity {

	@Override
	protected Class<? extends Fragment> getFragmentClass() {
		return UriMapperFragment.class;
	}

	@Override
	public UriHandler createUriHandler() {
		return new UriMapperUriHandler();
	}
}

File: jdroid-android-sample/src/main/java/com/jdroid/android/sample/ui/uri/UriMapperActivity.kt
package com.jdroid.android.sample.ui.uri

import androidx.fragment.app.Fragment

import com.jdroid.android.activity.FragmentContainerActivity
import com.jdroid.android.uri.UriHandler

class UriMapperActivity : FragmentContainerActivity() {

    override fun getFragmentClass(): Class<out Fragment> {
        return UriMapperFragment::class.java
    }

    override fun createUriHandler(): UriHandler<*>? {
        return UriMapperUriHandler()
    }
}